pkg/handlers/archive.go

```diff
@@ -108,11 +101,14 @@ func (a *Archive) openArchive(ctx context.Context, depth int, reader io.Reader,
 
 	switch archive := format.(type) {
 	case archiver.Decompressor:
-		info := decompressorInfo{depth: depth, reader: arReader, archiveChan: archiveChan, archiver: archive}
-
-		return a.handleDecompressor(ctx, info)
+		// Decompress tha archive and feed the decompressed data back into the archive handler to extract any nested archives.
+		compReader, err := archive.OpenReader(arReader)
+		if err != nil {
+			return err
+		}
+		return a.openArchive(ctx, depth+1, compReader, archiveChan)
 	case archiver.Extractor:
-		return archive.Extract(context.WithValue(ctx, depthKey, depth+1), reader, nil, a.extractorHandler(archiveChan))
+		return archive.Extract(context.WithValue(ctx, depthKey, depth+1), arReader, nil, a.extractorHandler(archiveChan))
```

Comment:
Why did this change?

Reply:
The extract should be using reader from `archiver.Identify` otherwise `Extract` is using the wrong reader. We should only be using the `arReader` once it's available.

Reply:
The reason why it currently works is because we are reading all the data out of the compressed file and passing the  `bytes.NewReader` back to `openArchive`. If we pass the reader directly, we need to make sure to use the new reader as a result of `archiver.Identify`.